exporter/loadbalancingexporter/helpers.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package loadbalancingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// mergeTraces concatenates multiple ptrace.Traces into a single ptrace.Traces.
+func mergeTraces(traces ...ptrace.Traces) ptrace.Traces {
+	merged := ptrace.NewTraces()
+	empty := ptrace.Traces{}
+
+	for _, trace := range traces {
+
+		if trace == empty {
+			continue
+		}
+
+		for i := 0; i < trace.ResourceSpans().Len(); i++ {
+			rs := trace.ResourceSpans().At(i)
+
+			newRS := merged.ResourceSpans().AppendEmpty()
+			rs.Resource().CopyTo(newRS.Resource())
+			newRS.SetSchemaUrl(rs.SchemaUrl())
+
+			for j := 0; j < rs.ScopeSpans().Len(); j++ {
+				ils := rs.ScopeSpans().At(j)
+
+				newILS := newRS.ScopeSpans().AppendEmpty()
+				ils.Scope().CopyTo(newILS.Scope())
+				newILS.SetSchemaUrl(ils.SchemaUrl())
+
+				for k := 0; k < ils.Spans().Len(); k++ {
+					span := ils.Spans().At(k)
+					newSpan := newILS.Spans().AppendEmpty()
+					span.CopyTo(newSpan)
+				}
+			}
+		}
```

Comment:
Is the pdata mutable? If yes, you should use `trace.ResourceSpans().At(i).MoveTo(merged.ResourceSpans().AppendEmpty())` or if not use `trace.ResourceSpans().At(i).CopyTo(merged.ResourceSpans().AppendEmpty())`